indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java

```diff
@@ -424,15 +426,16 @@ private void manageInternalCritical(
           catch (Exception e) {
             log.warn(e, "Exception thrown during isReady for task: %s", task.getId());
             final String errorMessage;
-            if (e instanceof MaxAllowedLocksExceededException) {
+            if (e instanceof MaxAllowedLocksExceededException || e instanceof LockAcquisitionFailedException) {
               errorMessage = e.getMessage();
             } else {
               errorMessage = StringUtils.format(
-                  "Encountered error[%s] while waiting for task to be ready. See Overlord logs for more details.",
-                  StringUtils.chop(e.getMessage(), 100)
+                  "Encountered error while waiting for task to be ready. See Overlord logs for more details. Error: %s",
+                  Throwables.getStackTraceAsString(e)
```

Comment:
`e.getMessage()` is probably enough, stacktrace is too long for task status reporting I think.

Reply:
+1, I prefer the old message here. (We can still remove the truncation limit of 100 though as it is already being handled inside `TaskStatus`.)

We are already giving out the info that the task could not become ready and some error occurred. That is enough info for an end user. After that, it is up to the admin/operator to look at the logs to know the complete stack trace.

Reply:
Fixed